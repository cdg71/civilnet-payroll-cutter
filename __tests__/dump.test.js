const fs = require("fs").promises;
const path = require("path");
const cuid = require("cuid");
const run = require("../utils/runner");
const { dumpFolder } = require("../src/lib/parser");

describe("En tant qu'utilisateur je veux extraire le contenu d'un train de paye au format PDF pour l'analyser. Le script devrait pouvoir...", () => {
  // On crée un dossier unique pour la suite de tests
  const testSuiteFolder = `./tests_results/dump_${cuid.slug()}`;

  // Avant les tests, crée le dossier de la suite de tests avant de l'utiliser ou on le vide s'il existe
  beforeAll(async () => {
    try {
      let exists = false;
      exists = await fs.stat(testSuiteFolder).catch(() => null);
      if (exists)
        await fs.rmdir(testSuiteFolder, {
          recursive: true,
        });
      await fs.mkdir(testSuiteFolder, {
        recursive: true,
      });
    } catch (error) {
      console.log(error);
    }
  });

  // Après les tests, on supprime le dossier de la suite de tests
  afterAll(async () => {
    try {
      let exists = false;
      exists = await fs.stat(testSuiteFolder).catch(() => null);
      if (exists)
        await fs.rmdir(testSuiteFolder, {
          recursive: true,
        });
    } catch (error) {
      console.log(error);
    }
  });

  test("extraire le contenu d'un fichier PDF avec une feuille de paye d'une page", async () => {
    expect.assertions(1);
    let result = null;
    try {
      const testFolder = path.resolve(testSuiteFolder, cuid.slug());
      const fixturePath = path.resolve(testFolder, "fixture.pdf");
      const dumpPath = path.resolve(testFolder, "dump.json");
      await run("./src/mock.js", ["-o", fixturePath]);
      await run("./src/dump.js", ["-i", fixturePath, "-o", dumpPath]);
      result = await dumpFolder(testFolder);
      expect(result).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });

  test("extraire le contenu d'un fichier PDF avec une feuille de paye de trois pages", async () => {
    expect.assertions(1);
    let result = null;
    try {
      const testFolder = path.resolve(testSuiteFolder, cuid.slug());
      const fixturePath = path.resolve(testFolder, "fixture.pdf");
      const dumpPath = path.resolve(testFolder, "dump.json");
      await run("./src/mock.js", ["-o", fixturePath, "-n", 3, "-p", "[1, 2]"]);
      await run("./src/dump.js", ["-i", fixturePath, "-o", dumpPath]);
      result = await dumpFolder(testFolder);
      expect(result).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });

  test("extraire le contenu d'un fichier PDF avec trois feuilles de paye : d'une page, de deux pages et d'une page", async () => {
    expect.assertions(1);
    let result = null;
    try {
      const testFolder = path.resolve(testSuiteFolder, cuid.slug());
      const fixturePath = path.resolve(testFolder, "fixture.pdf");
      const dumpPath = path.resolve(testFolder, "dump.json");
      await run("./src/mock.js", ["-o", fixturePath, "-n", 4, "-p", "[2]"]);
      await run("./src/dump.js", ["-i", fixturePath, "-o", dumpPath]);
      result = await dumpFolder(testFolder);
      expect(result).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });
});
