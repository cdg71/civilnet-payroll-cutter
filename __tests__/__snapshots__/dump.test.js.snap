// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`En tant qu'utilisateur je veux extraire le contenu d'un train de paye au format PDF pour l'analyser. Le script devrait pouvoir... extraire le contenu d'un fichier PDF avec trois feuilles de paye : d'une page, de deux pages et d'une page 1`] = `
Object {
  "cf3fd7e4c71c167bdd7bae14b184b297": Array [
    Array [
      "FAUX BULLETIN #1",
      "PAGE DU BULLETIN #1",
      "PAGE DU TRAIN DE PAYE #1",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM1",
      "11",
      "12",
      "NUMSECU1",
      "CLESECU1",
      "15",
      "16",
      "17",
    ],
    Array [
      "FAUX BULLETIN #2",
      "PAGE DU BULLETIN #1",
      "PAGE DU TRAIN DE PAYE #2",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM2",
      "11",
      "12",
      "NUMSECU2",
      "CLESECU2",
      "15",
      "16",
      ".../...",
    ],
    Array [
      "FAUX BULLETIN #2",
      "PAGE DU BULLETIN #2",
      "PAGE DU TRAIN DE PAYE #3",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM2",
      "11",
      "12",
      "NUMSECU2",
      "CLESECU2",
      "15",
      "16",
      "17",
    ],
    Array [
      "FAUX BULLETIN #3",
      "PAGE DU BULLETIN #1",
      "PAGE DU TRAIN DE PAYE #4",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM3",
      "11",
      "12",
      "NUMSECU3",
      "CLESECU3",
      "15",
      "16",
      "17",
    ],
  ],
}
`;

exports[`En tant qu'utilisateur je veux extraire le contenu d'un train de paye au format PDF pour l'analyser. Le script devrait pouvoir... extraire le contenu d'un fichier PDF avec une feuille de paye d'une page 1`] = `
Object {
  "c6c15e992c218e8a9138a04e0cb0c27f": Array [
    Array [
      "FAUX BULLETIN #1",
      "PAGE DU BULLETIN #1",
      "PAGE DU TRAIN DE PAYE #1",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM1",
      "11",
      "12",
      "NUMSECU1",
      "CLESECU1",
      "15",
      "16",
      "17",
    ],
  ],
}
`;

exports[`En tant qu'utilisateur je veux extraire le contenu d'un train de paye au format PDF pour l'analyser. Le script devrait pouvoir... extraire le contenu d'un fichier PDF avec une feuille de paye de trois pages 1`] = `
Object {
  "dabedf944505b06aa3a7b889e3ba0231": Array [
    Array [
      "FAUX BULLETIN #1",
      "PAGE DU BULLETIN #1",
      "PAGE DU TRAIN DE PAYE #1",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM1",
      "11",
      "12",
      "NUMSECU1",
      "CLESECU1",
      "15",
      "16",
      ".../...",
    ],
    Array [
      "FAUX BULLETIN #1",
      "PAGE DU BULLETIN #2",
      "PAGE DU TRAIN DE PAYE #2",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM1",
      "11",
      "12",
      "NUMSECU1",
      "CLESECU1",
      "15",
      "16",
      ".../...",
    ],
    Array [
      "FAUX BULLETIN #1",
      "PAGE DU BULLETIN #3",
      "PAGE DU TRAIN DE PAYE #3",
      "4",
      "5",
      "6",
      "7",
      "01-01-2020 - 31-01-2020",
      "9",
      "NOM1",
      "11",
      "12",
      "NUMSECU1",
      "CLESECU1",
      "15",
      "16",
      "17",
    ],
  ],
}
`;
